src/artikler/sok-sosialhjelp/komponenter/kommunesok/service/useTilgjengeligeKommunerService.ts

```diff
@@ -68,33 +80,37 @@ const useTilgjengeligeKommunerService = () => {
         };
         fetch(url, options)
             .then((response) => response.json())
-            .then((response) =>
+            .then((response: {[key: string]: KommuneInfo}) => {
                 setResult({
                     restStatus: REST_STATUS.OK,
                     payload: {results: response},
-                })
-            )
+                });
+            })
             .catch((error) =>
                 setResult({restStatus: REST_STATUS.FEILET, error})
             );
     }, [url]);
+    console.log("result", result);
     return result;
 };
 
 const finnTilgjengeligKommune = (
-    tilgjengeligeKommuner: string[],
+    tilgjengeligeKommuner: {[key: string]: KommuneInfo},
     kommunenummer: string
-): boolean => {
-    let funnetKommune: any = undefined;
-    tilgjengeligeKommuner.map((tilgjengeligeKommuneNr: string) => {
-        if (tilgjengeligeKommuneNr.match(kommunenummer)) {
-            // console.log("bingo!: " + tilgjengeligeKommuneNr + ".match( " + kommunenummer);
-            funnetKommune = tilgjengeligeKommuneNr;
-        }
-        return funnetKommune;
-    });
+): KommuneInfo | undefined => {
+    const tilgjengeligKommune = Object.values(
+        tilgjengeligeKommuner
+    ).filter((kommune) => kommune.kommunenummer.match(kommunenummer));
+
+    return tilgjengeligKommune[0];
```

Comment:
Er dette faktisk trygt? Regner jo med at du vet bedre enn meg ;-)

Reply:
Kunne nok vært skrevet bedre.. Men koden returnerer en kommune hvis den finner en, og undefined hvis ikke. Tar gjerne i mot innspill på forbedringer. :)

Reply:
I java så er jo dette en klassisk index out of bounds, men greit nok så lenge det funker.

Reply:
Jeg skrev om til noe som forhåpentligvis virker litt mer lettlest. Kan nok komme tilbake til om det finnes bedre måter å løse dette på.

```typescript
const finnTilgjengeligKommune = (
    tilgjengeligeKommuner: {[key: string]: KommuneInfo},
    kommunenummer: string
): KommuneInfo | undefined => {
    return Object.values(tilgjengeligeKommuner)
        .filter((kommune) => kommune.kommunenummer.match(kommunenummer))
        .shift();
};
```

Opprinnelig hadde vi bare en boolean som sa om en valgt kommune hadde digital søknad eller ikke. Det skalerte ikke så bra når vi også skulle ta med status på innsyn.

Reply:
Begge returnerer undefined (i chrome console) og sender ikke ut nullpointerException som vi er vandt med fra backend. 
![image](https://user-images.githubusercontent.com/4296877/96147474-a7d6ec00-0f07-11eb-8709-a811bcdc66ac.png)